File: src/main/java/org/labun/jooq/codegen/util/DefaultNameCreator.java
package org.labun.jooq.codegen.util;

import lombok.Getter;
import lombok.Setter;
import org.apache.commons.lang3.StringUtils;
import org.jooq.util.Definition;
import org.labun.jooq.codegen.config.CodeGenerationConfig;
import org.labun.jooq.codegen.config.Configuration;
import org.labun.jooq.codegen.config.NameConfig;

/**
 * @author Konstantin Labun
 */
@Getter
@Setter
public class DefaultNameCreator implements NameCreator, Configurable {
    private static final CodeGenerationConfig DEFAULT = defaultCodeGenerationConfig();

    private TableNameConverter nameConverter = new TableNameConverter();
    private Configuration configuration;

    private static CodeGenerationConfig defaultCodeGenerationConfig() {
        return new CodeGenerationConfig();
    }

    @Override
    public String createClassName(CodeGenerationConfig cfg, Definition table) {
        NameConfig name = cfg.className();
        return name.prefix() + nameConverter.convert(table) + name.postfix();
    }

    @Override
    public String createGetterName(CodeGenerationConfig cfg, Definition column) {
        NameConfig name = cfg.accessors().getters();

        String result = name.prefix() + nameConverter.convert(column) + name.postfix();
        return StringUtils.uncapitalize(result);
    }

    @Override
    public String createSetterName(CodeGenerationConfig cfg, Definition column) {
        NameConfig name = cfg.accessors().setters();

        String result = name.prefix() + nameConverter.convert(column) + name.postfix();
        return StringUtils.uncapitalize(result);
    }

    @Override
    public void configure(Configuration configuration) {
        this.configuration = configuration;
    }
}

File: src/main/kotlin/org/labun/jooq/generator/util/DefaultNameCreator.kt
package org.labun.jooq.generator.util

import lombok.Getter
import lombok.Setter
import org.apache.commons.lang3.StringUtils
import org.jooq.util.Definition
import org.labun.jooq.generator.config.SubGeneratorConfig
import org.labun.jooq.generator.config.Configuration

/**
 * @author Konstantin Labun
 */
@Getter
@Setter
class DefaultNameCreator : NameCreator, Configurable {

  private val nameConverter = TableNameConverter()
  private var configuration: Configuration? = null

  override fun createClassName(cfg: SubGeneratorConfig, table: Definition): String {
    val name = cfg.className
    return name.prefix + nameConverter.convert(table) + name.postfix
  }

  override fun createFieldName(cfg: SubGeneratorConfig, column: Definition): String {
    return nameConverter.convert(column)
  }

  override fun createGetterName(cfg: SubGeneratorConfig, column: Definition): String {
    val name = cfg.accessors.getters

    val result = name.prefix + nameConverter.convert(column) + name.postfix
    return StringUtils.uncapitalize(result)
  }

  override fun createSetterName(cfg: SubGeneratorConfig, column: Definition): String {
    val name = cfg.accessors.setters

    val result = name.prefix + nameConverter.convert(column) + name.postfix
    return StringUtils.uncapitalize(result)
  }

  override fun configure(configuration: Configuration) {
    this.configuration = configuration
  }

  companion object {
    private val DEFAULT = defaultCodeGenerationConfig()

    private fun defaultCodeGenerationConfig(): SubGeneratorConfig {
      return SubGeneratorConfig()
    }
  }
}
